device_doctor/lib/src/process_helper.dart

```diff
@@ -67,3 +61,56 @@ Future<String> eval(String executable, List<String> arguments,
 
   return output.trimRight();
 }
+
+/// Splits [from] into lines and selects those that contain [pattern].
+Iterable<String> grep(Pattern pattern, {@required String from}) {
+  return from.split('\n').where((String line) {
+    return line.contains(pattern);
+  });
+}
+
+Future<bool> killAllRunningProcessesOnWindows(String processName, {ProcessManager processManager}) async {
+  processManager ??= LocalProcessManager();
+  // Avoid endless loop when a process from a different use exists, and fails
+  // to get killed every try.
+  int tries = 3;
```

Comment:
use retry package instead of implementing your own logic?

Reply:
retry packages is suitable for retry logics with exception. Here we simply keep killing process if exist.